src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,489 @@
+        see if this exists and if not, will raise an exception.
+    incW: bool
+        If incW is True, then the back trajectories will take into account vertical velocity fields
+        and thus a back trajectory takes a 3D path back. In this case, vertical wind fields are
+        smoothed in addition to the prior coarsening, to prevent instantaneous sampling of strong
+        updrafts/downdrafts that would advect a parcel to the bottom/top of atmosphere if the
+        temporal sampling of the data is coarse enough (e.g. hourly or 3 hourly). This option enables
+        the trajectory to capture large areas of rising/sinking air, in phenomena such as overturning
+        circulations. If incW if False, then compute back trajectories on a pressure level with no
+        vertical advection (essentially a laminar, stratified flow).
+    cyclic: bool
+        If cyclic is True, then the code will assume no east/west boundary and if a back trajectory
+        reaches the boundary, it will emerge out of the other side. This option is useful for large
+        domains such as the K-SCALE tropical channel, where there are only north/south boundaries in
+        the domain.
+    multicore: bool
+        If true, split up age of air diagnostic to use multiple cores (defaults to 8), otherwise run
+        using a single process, which is easier to debug.
+
+    Returns
+    -------
+    Cube
+        An iris cube of the age of air data, with 3 dimensions (time, latitude, longitude).
+
+    """
+    # Set up temporary directory to store intermediate age of air slices.
+    tmpdir = tempfile.TemporaryDirectory()
+    logging.info("Made tmpdir %s", tmpdir.name)
+
+    # Check that all cubes are of same size (will catch different dimension orders too).
+    if not XWIND.shape == YWIND.shape == WWIND.shape == GEOPOT.shape:
+        raise ValueError("Cubes are not the same shape")
+
+    # Get time units and assign for later
+    if str(XWIND.coord("time").units).startswith("hours since "):
+        timeunit = "hour"
+    else:
+        raise NotImplementedError("Unsupported time base")
+
+    # Make data non-lazy to speed up code.
+    logging.info("Making data non-lazy...")
+    x_arr = XWIND.data
+    y_arr = YWIND.data
+    z_arr = WWIND.data
+    g_arr = GEOPOT.data
+
+    # Get coord points
+    lat_name, lon_name = get_cube_yxcoordname(XWIND)
+    lats = XWIND.coord(lat_name).points
+    lons = XWIND.coord(lon_name).points
+    time = XWIND.coord("time").points
+
+    # Get time spacing of cube to determine whether the spacing in time is the
+    # same throughout the cube. If not, then not supported.
+    dt = XWIND.coord("time").points[1:] - XWIND.coord("time").points[:-1]
+    if np.all(dt == dt[0]):
+        dt = dt[0]
+    else:
+        raise NotImplementedError("Time intervals are not consistent")
+
+    # Some logic to determine which index each axis is, and check for ensembles.
+    dimension_mapping = {}
```

Comment:
Given we have this dimension mapping, I feel that is should be possible to support cubes with different dimension layouts. I'm not sure how common that would be however, so it might not be worth it.